scripts/confix/confix_test.go

```diff
@@ -0,0 +1,94 @@
+package main_test
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/creachadair/tomledit"
+	"github.com/google/go-cmp/cmp"
+	confix "github.com/tendermint/tendermint/scripts/confix"
+)
+
+func mustParseConfig(t *testing.T, path string) *tomledit.Document {
+	doc, err := confix.LoadConfig(path)
+	if err != nil {
+		t.Fatalf("Loading config: %v", err)
+	}
+	return doc
+}
+
+func TestGuessConfigVersion(t *testing.T) {
```

Comment:
I'd recommend also trying to throw in a garbled version, like a toml file that still contains an old field that may not have caused a newer version of tendermint to barf.

Reply:
I'm not sure I have a good example of that. The great renaming of `snake_case` to `kebab-case` in v0.35 means most anything from older versions is invalid. Do you have a suggestion for an example?

Reply:
_sigh_ I suppose I don't. I was thinking we could grep around for some, but since most chains are on v0.34 we are unlikely to find any.

Reply:
It seems like a good idea, I just didn't have any luck thinking of examples. 

The testdata files I have in this PR are just the "default config" emitted from each of the versions of Tendermint (getting some of the older ones to build at all was a bit of an adventure, so I didn't do much else).